x-pack/plugins/cross_cluster_replication/public/app/services/routing.js

```diff
@@ -68,7 +68,7 @@ class Routing {
       : undefined;
 
     this._reactRouter.history.push({
-      pathname: appToBasePathMap[app] + route,
+      pathname: encodeURI(appToBasePathMap[app] + route),
```

Comment:
In which situations is this necessary?

Reply:
Whenever we navigate to a route that might contain special characters. Like when editing an auto-follow pattern with this name :`%&()&"` 😊 The url `/auto_follow_patterns/edit/<name>` needs to be encoded

Reply:
Aren't we already encoding the name when we call `navigate`? e.g. https://github.com/sebelga/kibana/pull/5/files#diff-5ec658710f360dfa483d45245036ba62R150